credentials/src/aws_clientcredential_keys.c

```diff
@@ -0,0 +1,155 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ * http://aws.amazon.com/freertos
+ * http://www.FreeRTOS.org
+ */
+
+#include <stdint.h>
+#include <stddef.h>
+
+/*
+ * @brief PEM-encoded client certificate.
+ *
+ * @todo If you are running one of the Amazon FreeRTOS demo projects, set this
+ * to the certificate that will be used for TLS client authentication.
+ *
+ * @note Must include the PEM header and footer:
+ * "-----BEGIN CERTIFICATE-----\n"\
+ * "...base64 data...\n"\
+ * "-----END CERTIFICATE-----\n"
+ */
+#define keyCLIENT_CERTIFICATE_PEM ""
+
+/*
+ * @brief PEM-encoded issuer certificate for AWS IoT Just In Time Registration
+ * (JITR).
+ *
+ * @todo If you are using AWS IoT Just in Time Registration (JITR), set this to
+ * the issuer (Certificate Authority) certificate of the client certificate
+ * above.
+ *
+ * @note This setting is required by JITR because the issuer is used by the AWS
+ * IoT gateway for routing the device's initial request. (The device client
+ * certificate must always be sent as well.) For more information about JITR,
+ * see:
+ *  https://docs.aws.amazon.com/iot/latest/developerguide/jit-provisioning.html,
+ *  https://aws.amazon.com/blogs/iot/just-in-time-registration-of-device-certificates-on-aws-iot/.
+ *
+ * If you're not using JITR, set below to NULL.
+ *
+ * Must include the PEM header and footer:
+ * "-----BEGIN CERTIFICATE-----\n"\
+ * "...base64 data...\n"\
+ * "-----END CERTIFICATE-----\n"
+ */
+#define keyJITR_DEVICE_CERTIFICATE_AUTHORITY_PEM ""
+
+/*
+ * @brief PEM-encoded client private key.
+ *
+ * @todo If you are running one of the Amazon FreeRTOS demo projects, set this
+ * to the private key that will be used for TLS client authentication.
+ *
+ * @note Must include the PEM header and footer:
+ * "-----BEGIN RSA PRIVATE KEY-----\n"\
+ * "...base64 data...\n"\
+ * "-----END RSA PRIVATE KEY-----\n"
+ */
+#define keyCLIENT_PRIVATE_KEY_PEM ""
+
+const char * pucGetClientCert( uint32_t* ulCertSize )
+{
+    if ( ulCertSize != NULL ) 
+	{
+        *ulCertSize = sizeof( keyCLIENT_CERTIFICATE_PEM );
+    }
+
+    return keyCLIENT_CERTIFICATE_PEM;
+}
+
+const char * pucGetJITPCert( uint32_t* ulCertSize )
+{
+
+    if ( ulCertSize != NULL ) 
+	{
+        *ulCertSize = sizeof( keyJITR_DEVICE_CERTIFICATE_AUTHORITY_PEM );
+    }
+
+    return keyJITR_DEVICE_CERTIFICATE_AUTHORITY_PEM;
+}
+
+const char * pucGetClientKey( uint32_t* ulKeySize )
+{
+    if ( ulKeySize != NULL ) 
+	{
+        *ulKeySize = sizeof( keyCLIENT_PRIVATE_KEY_PEM );
+    }
+
+    return keyCLIENT_PRIVATE_KEY_PEM;
+}
+
+void vProvisionClientCert()
```

Comment:
provision seems to be a bit of an overloaded name.... Would love suggestions.
Perhaps vLoad? or vStore?

Reply:
Agreed. Your comment below says "load", so why not go with that inspiration?

Reply:
Awesome